[AArch64] Improve bf16 fp_extend lowering.

davemgreen opened the pull request:
A bf16 fp_extend is just a shift into the higher bits. This changes the lowering from using a relatively ugly tablegen pattern, to ISel generating the shift using an extended vector. This is cleaner and should optimize better. StrictFP goes through the same route as it cannot round or set flags.

nasherm commented on llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:
```diff
@@ -753,6 +753,14 @@ AArch64TargetLowering::AArch64TargetLowering(const TargetMachine &TM,
     setOperationAction(Op, MVT::v8bf16, Expand);
   }
 
+  // For bf16, fpextend is custom lowered to optionally expanded into shifts.
```
Nit: typo I think. `lowered to optionally` -> `lowered to be optionally`

nasherm reviewed (approved):
LGTM. Just one nit

davemgreen committed:
[AArch64] Improve bf16 fp_extend lowering.

A bf16 fp_extend is just a shift into the higher bits. This changes the
lowering from using a relatively ugly tablegen pattern, to ISel generating the
shift using an extended vector. This is cleaner and should optimize better.
StrictFP goes through the same route as it cannot round or set flags.
